galaxy_ng/app/auth/ldap.py

```diff
@@ -40,3 +40,22 @@ class GalaxyLDAPBackend(LDAPBackend):
 
     def __init__(self):
         self.settings = GalaxyLDAPSettings(self.settings_prefix, self.default_settings)
+
+
+class PrefixedLDAPBackend(GalaxyLDAPBackend):
+
+    @property
+    def prefix(self):
+        return settings.get("RENAMED_USERNAME_PREFIX")
+
+    def authenticate(self, *args, **kwargs):
+        if username := kwargs.get("username"):
+            if username.startswith(self.prefix):
+                kwargs["username"] = username[len(self.prefix):]
+                return super().authenticate(*args, **kwargs)
```

Comment:
Python 3.9 added a removeprefix to `str` https://docs.python.org/3.9/library/stdtypes.html#str.removeprefix 

```suggestion
        if username := kwargs.get("username"):
            kwargs["username"] = username[len(self.prefix):]
            return super().authenticate(*args, **kwargs)
```

Reply:
This goes the opposite direction of the DAB mixin.

https://github.com/ansible/django-ansible-base/blob/devel/ansible_base/lib/backends/prefixed_user_auth.py

That adds the prefix to the username, as opposed to removing it. I can't find an explanation for why galaxy_ng specifically is doing this.

Reply:
On LDAP users has no prefix, then the LDAP backend must remove it to succed the search?

Reply:
In the migration, a user "rick" is a _conflict_ because it exists both in controller & Hub. So the user in Hub will be renamed to "hub_rick", and "rick" will track the user from controller.

Scenario we care about is that someone logs in with "rick" username and the password which now only works for "hub_rick". So in the case of password auth, we would have to add the prefix to the username.

In the case of social auth, the client would still give "rick" username and expect it to work. Of course, this user will only be known as "rick" to the external auth provider. But IIRC, @newswangerd 's code sample still ran the same prefixing logic as password auth. I agree that might not make complete sense, but this diff doesn't make sense to me either.